test-integration/repay-loan/1_repay_loan_3_payments.js

```diff
@@ -0,0 +1,176 @@
+
+  // Deposit tokens on lending pool.
+  console.log('Depositing tokens on lending pool...');
+  const lenderTxConfig = await accounts.getTxConfigAt(0);
+  await token.approve(lendingPoolInstance.address, lendingPoolDepositAmountWei, lenderTxConfig);
+  const depositResult = await lendingPoolInstance.deposit(lendingPoolDepositAmountWei, lenderTxConfig);
+  lendingPool
+    .tokenDeposited(depositResult)
+    .emitted(lenderTxConfig.from, lendingPoolDepositAmountWei);
+
+  // Set loan terms.
+  console.log('Setting loan terms...');
+  const loanTermsRequestInfo = {
+    borrower,
+    recipient,
+    requestNonce: nonces.newNonce(borrower),
+    amount: amountWei.toFixed(0),
+    duration: durationInDays * ONE_DAY,
+    requestTime: currentTimestamp,
+    caller: loansInstance.address,
+    consensusAddress: loanTermConsensusInstance.address,
+  };
+  const loanResponseInfoTemplate = {
+    responseTime: currentTimestamp - 10,
+    interestRate: 400,
+    collateralRatio: 600,
+    maxLoanAmount: maxAmountWei.toFixed(0),
+    consensusAddress: loanTermConsensusInstance.address,
+  };
+  const loanTermsRequest = createLoanTermsRequest(loanTermsRequestInfo, chainId);
+  const signedResponses = await createMultipleSignedLoanTermsResponses(
+    web3,
+    loanTermsRequest,
+    signers,
+    loanResponseInfoTemplate,
+    nonces,
+    chainId,
+  );
+
+  const createLoanWithTermsResult = await loansInstance.createLoanWithTerms(
+    loanTermsRequest.loanTermsRequest,
+    signedResponses,
+    borrowerTxConfigWithValue.value,
+    borrowerTxConfigWithValue
+  );
+
+  const termsExpiryTime = await settingsInstance.termsExpiryTime();
+  const expiryTermsExpected = await timer.getCurrentTimestampInSecondsAndSum(termsExpiryTime);
+  const loanIDs = await loansInstance.getBorrowerLoans(borrower);
+  const lastLoanID = loanIDs[loanIDs.length - 1];
+  loans
+    .loanTermsSet(createLoanWithTermsResult)
+    .emitted(
+      lastLoanID,
+      borrowerTxConfigWithValue.from,
+      recipient,
+      loanResponseInfoTemplate.interestRate,
+      loanResponseInfoTemplate.collateralRatio,
+      loanResponseInfoTemplate.maxLoanAmount,
+      loanTermsRequestInfo.duration,
+      expiryTermsExpected,
+    );
+
+  console.log(`Advancing time to take out loan (current: ${(await timer.getCurrentDate())})...`);
+  const nextTimestamp = await timer.getCurrentTimestampInSecondsAndSum(minutesToSeconds(2));
+  await timer.advanceBlockAtTime(nextTimestamp);
+  
+  // Take out a loan.
+  console.log(`Taking out loan id ${lastLoanID}...`);
+  const takeOutLoanResult = await loansInstance.takeOutLoan(lastLoanID, amountWei, borrowerTxConfig);
+  loans
+    .loanTakenOut(takeOutLoanResult)
+    .emitted(lastLoanID, borrowerTxConfig.from, amountWei);
+
+  // Set a lower price for Token/ETH.
+  console.log(`Settings final (lower) oracle price: 1 ${tokenName} = ${finalOraclePrice.toFixed(0)} WEI = ${toUnits(finalOraclePrice, 18)} ETHER`);
```

Comment:
I think this new oracle price is not needed in the scenario.

Reply:
Removed

Reply:
I think it wasn't removed.

Reply:
I think it wasn't removed. https://github.com/teller-protocol/teller-protocol-v1/blob/7dbb5ccbfaeb02ead9963a336117aa81f0869b70/test-integration/repay-loan/1_repay_loan_3_payments.js#L116